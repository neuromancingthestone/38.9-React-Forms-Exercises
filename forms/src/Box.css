.Box {
  margin: 0 auto;
}