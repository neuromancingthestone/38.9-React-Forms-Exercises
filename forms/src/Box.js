import "./Box.css"

const Box = ({id, color, width, height, remove}) => {
  return (
    <div 
      className='Box'
      id={id}
      style={{
        backgroundColor: color,
        width: width,
        height: height
      }}>
      <button onClick={remove}>X</button>
    </div>
  )
}

export default Box;