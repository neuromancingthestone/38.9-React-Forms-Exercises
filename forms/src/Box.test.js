import {act} from "react";
import { render, fireEvent } from "@testing-library/react";
import Box from "./Box";

test('It renders without crashing', () => {
  render(<Box 
    id={1}
    color={'teal'}
    width={'50px'}
    height={'50px'}
  />)
})

test('It matches snapshot', () => {
  const {asFragment} = render(<Box 
    id={1}
    color={'teal'}
    width={'50px'}
    height={'50px'}  
  />)
  expect(asFragment()).toMatchSnapshot();
})