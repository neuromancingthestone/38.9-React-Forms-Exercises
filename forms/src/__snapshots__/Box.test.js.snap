// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    id="1"
    style="background-color: teal; width: 50px; height: 50px;"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
